docs/index-server.md

```diff
@@ -0,0 +1,231 @@
+.inmemory.serialization.threshold.inKB'. By default, the minimum value for this property is 0,
+meaning that no matter how small the splits are they would be written to the file. Maximum is
+102400KB which will mean if the size of the splits for a executor cross this value then they would
+be written to file.
+
+The user can set th location for these file by using 'carbon.indexserver.temp.path'. By default
+table path would be used to write the files.
+
+## Configurations
+
+##### carbon.properties(JDBCServer) 
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| carbon.enable.index.server       |  false | Enable the use of index server for pruning for the whole application.       |
+| carbon.index.server.ip |    NA   |   Specify the IP/HOST on which the server would be started. Better to specify the private IP. | 
+| carbon.index.server.port | NA | The port on which the index server has to be started. |
+| carbon.disable.index.server.fallback | false | Whether to enable/disable fallback for index server
+. Should be used for testing purposes only. |
+|carbon.index.server.max.worker.threads| 500 | Number of RPC handlers to open for accepting the
+requests from JDBC driver. Max accepted value is Integer.Max. |
+|carbon.index.server.max.jobname.length|NA|The max length of the job to show in the index server application UI. For bigger queries this may impact performance as the whole string would be sent from JDBCServer to IndexServer.|
+
+
+##### carbon.properties(IndexServer) 
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| carbon.enable.index.server       |  false | Enable the use of index server for pruning for the
+whole application.       |
+| carbon.index.server.ip |    NA   |   Specify the IP/HOST on which the server would be started. Better to specify the private IP. | 
+| carbon.index.server.port | NA | The port on which the index server has to be started. |
+|carbon.index.server.max.worker.threads| 500 | Number of RPC handlers to open for accepting the
+requests from JDBC driver. Max accepted value is Integer.Max. |
+|carbon.max.executor.lru.cache.size|  NA | Used to specify the max size for executor LRU cache. Mandatory to set fo the user. |
+|carbon.index.server.max.jobname.length|NA|The max length of the job to show in the index server application UI. For bigger queries this may impact performance as the whole string would be sent from JDBCServer to IndexServer.|
+|carbon.max.executor.threads.for.block.pruning|4| max executor threads used for block pruning. |
+|carbon.index.server.inmemory.serialization.threshold.inKB|300|Max in memory serialization size
+after reaching threshold data will be written to file. Min value that the user can set is 0KB and
+max is 102400KB. |
+|carbon.indexserver.temp.path|tablePath|will be used to write split serialize data when in memory
+threashold crosses the limit|
+
+
```

Comment:
Can you add some information about how secure mode works for index server?

Reply:
added